tests/acl/test_acl.py

```diff
@@ -156,7 +156,7 @@ def acl_table_config(duthost, setup, stage):
     }
 
     logger.info('extra variables for ACL table:\n{}'.format(pprint.pformat(acl_table_vars)))
-    duthost.host.options['variable_manager'].extra_vars = acl_table_vars
+    duthost.host.options['variable_manager']._extra_vars = acl_table_vars
```

Comment:
for this one, wonder if it is better to use extra_vars.clear() and then extra_vars.update(acl_table_vars) is better? feel _extra_vars is a private object. better to operate on extra_vars object directly.

Reply:
I see many other similiar cases, should we change all of them?

```
tests/acl/test_acl.py:    duthost.host.options['variable_manager'].extra_vars = acl_table_vars
tests/test_vrf.py:    extra_vars = {'cfg_t0': cfg_t0,
tests/test_vrf.py:    duthost.host.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf.py:    extra_vars = {
tests/test_vrf.py:    ptfhost.host.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf.py:    extra_vars = {
tests/test_vrf.py:    ptfhost.host.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf.py:        extra_vars = {
tests/test_vrf.py:        duthost.host.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf.py:        extra_vars = {
tests/test_vrf.py:        ptfhost.host.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf.py:        dut_extra_vars = {
tests/test_vrf.py:        duthost.host.options['variable_manager'].extra_vars = dut_extra_vars
tests/test_vrf.py:        ptf_extra_vars = {
tests/test_vrf.py:        ptfhost.host.options['variable_manager'].extra_vars = ptf_extra_vars
tests/test_vrf.py:        dut_extra_vars.update({
tests/test_vrf.py:        duthost.host.options['variable_manager'].extra_vars = dut_extra_vars
tests/test_vrf.py:            dut_extra_vars['op_code'] = 'del'
tests/test_vrf.py:            duthost.host.options['variable_manager'].extra_vars = dut_extra_vars
tests/test_vrf.py:        dut_extra_vars = {
tests/test_vrf.py:        duthost.host.options['variable_manager'].extra_vars = dut_extra_vars
tests/test_vrf_attr.py:        extra_vars = { 'router_mac': self.new_vrf1_router_mac }
tests/test_vrf_attr.py:        duthost.options['variable_manager'].extra_vars = extra_vars
tests/test_vrf_attr.py:        extra_vars = { 'router_mac': host_facts['ansible_Ethernet0']['macaddress'] }
tests/test_vrf_attr.py:        duthost.host.options['variable_manager'].extra_vars = extra_vars
```

Reply:
In ansible 2.8.7, extra_vars is exposed as a getter only property. The workaround is to directly access the "private" attribute `_extra_vars`.  So, `extra_vars.clear()` and `extra_vars.update()` won't work. `_extra_vars.clear()` and `_extra_vars.update()` will work.

Reply:
Actually `extra_vars.clear()` and  `extra_vars.update()` will work as `extra_vars` returns referens to the same dict object as `_extra_vars`.  Example: https://onlinegdb.com/BynYjKJ0r

Reply:
@lguohan Fixed. According to https://docs.python.org/3/library/stdtypes.html#dict.update there is no need to call `clear()` method.